app/javascript/components/tooltips/Student.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { useRequestQuery } from '../../hooks/request-query'
+import { QueryResult } from 'react-query'
+import { fromNow } from '../../utils/time'
+import { Loading } from '../common/Loading'
+
+export function Student({
+  endpoint,
+  styles,
+}: {
+  endpoint: string
+  styles?: React.CSSProperties
+}) {
+  const request = {
+    endpoint: endpoint,
+    options: { staleTime: 1000 * 60 },
+  }
+  const { isLoading, isError, isSuccess, data } = useRequestQuery(
+    'student-tooltip',
+    request
+  ) as QueryResult<any>
```

Comment:
The data here is `any`, but then it's destructured on line 27. That's probably a bad idea.

Reply:
It is. This was also temporary, as I had a typing issue that I failed to solve. I've now pushed a new version that introduces some nice typing for the `useRequestQuery` function.